web/src/message_events.ts

```diff
@@ -520,7 +551,14 @@ export function update_messages(events) {
                      * history events. This logic ensures that all
                      * messages that were moved are displayed as such
                      * without a browser reload. */
-                    const edit_history_entry = {
+                    const edit_history_entry: {
+                        user_id: number | null;
+                        timestamp: number;
+                        stream?: number;
+                        prev_stream?: number;
+                        topic?: string;
+                        prev_topic?: string;
+                    } = {
```

Comment:
Probably a good follow-up would be to refine this type to be a union of different types of edit history event types used elsewhere in the codebase. `EditHistoryEntry` is fairly different. Might be appropriate to just leave an issue for this if it seems like work.